src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -110,51 +110,58 @@ public virtual bool TryCreate(FunctionMetadata functionMetadata, out FunctionDes
                 throw new ArgumentNullException("methodAttributes");
             }
 
+            ApplyMethodLevelAttributes(functionMetadata, triggerMetadata, methodAttributes);
+
+            Collection<ParameterDescriptor> parameters = new Collection<ParameterDescriptor>();
+            ParameterDescriptor triggerParameter = CreateTriggerParameter(triggerMetadata);
+            parameters.Add(triggerParameter);
+
+            // Add a TraceWriter for logging
+            parameters.Add(new ParameterDescriptor("log", typeof(TraceWriter)));
+
+            // Add an IBinder to support the binding programming model
+            parameters.Add(new ParameterDescriptor("binder", typeof(IBinder)));
+
+            // Add ExecutionContext to provide access to InvocationId, etc.
+            parameters.Add(new ParameterDescriptor("context", typeof(ExecutionContext)));
+
+            return parameters;
+        }
+
+        protected virtual ParameterDescriptor CreateTriggerParameter(BindingMetadata triggerMetadata, Type parameterType = null)
+        {
             ParameterDescriptor triggerParameter = null;
             switch (triggerMetadata.Type)
             {
                 case BindingType.QueueTrigger:
-                    triggerParameter = ParseQueueTrigger((QueueBindingMetadata)triggerMetadata);
+                    triggerParameter = ParseQueueTrigger((QueueBindingMetadata)triggerMetadata, parameterType);
```

Comment:
Because the other `Parse*Trigger` methods were also applying a default and had the type was optional, for consistency, the change I made now was to be explicit about the type. We can go further if we want, but I have a feeling that will be changing anyway was we work on the extensibility story.